content/en/spec/activitypub.md

```diff
@@ -357,203 +518,234 @@ Mastodon allows users to feature specific hashtags on their profile for easy bro
 
-Mastodon supports setting a focal point on uploaded images, so that wherever that image is displayed, the focal point stays in view. This is implemented using an extra property `focalPoint` on `Image` objects. The property is an array of two floating points between -1.0 and 1.0, with 0,0 being the center of the image, the first value being x (-1.0 is the left edge, +1.0 is the right edge) and the second value being y (-1.0 is the bottom edge, +1.0 is the top edge). See [API Guidelines > Focal points]({{< relref "api/guidelines#focal-points" >}}) for more information. Example:
+Mastodon supports integration with identity providers to prove that a profile is linked to a certain identity. This is implemented using the `attachment` property on actor objects, with objects in the array having a type of `IdentityProof` from the Mastodon namespace. The object also includes `signatureAlgorithm` and `signatureValue` from the W3ID Security Vocabulary namespace.
 
 ```json
 {
   "@context": [
     "https://www.w3.org/ns/activitystreams",
+    "https://w3id.org/security/v1",
     {
-      "toot": "http://joinmastodon.org/ns#",
-      "focalPoint": {
-        "@container": "@list",
-        "@id": "toot:focalPoint"
-      }
+      "IdentityProof": "http://joinmastodon.org/ns#IdentityProof"
     }
   ],
-
-  "id": "https://example.com/@alice/hello-world",
-  "type": "Note",
-  "content": "A picture attached!",
+  "id": "https://mastodon.social/users/Gargron",
+  "type": "Person",
   "attachment": [
     {
-      "type": "Image",
-      "mediaType": "image/png",
-      "url": "https://example.com/files/cats.png",
-      "focalPoint": [
-        -0.55,
-        0.43
-      ]
+      "type": "IdentityProof",
+      "name": "gargron",
+      "signatureAlgorithm": "keybase",
+      "signatureValue": "5cfc20c7018f2beefb42a68836da59a792e55daa4d118498c9b1898de7e845690f"
     }
   ]
 }
 ```
 
-{{< figure src="assets/focal-points.jpg" caption="A demonstration of various focal points and their coordinates." >}}
-
-The focal point of (-0.55, 0.43) in the example above corresponds to a point 55% to the left of center and 43% above center. This focal point should remain visible within the cropped thumbnail, if any cropping is done.
-
-### Blurhash {#blurhash}
+### Discoverability flag {#discoverable}
 
-Mastodon generates colorful preview thumbnails for attachments. This is implemented using an extra property `blurhash` on `Image` objects. The property is a string generated by the [BlurHash algorithm](https://blurha.sh). Example:
+Mastodon allows users to opt-in or opt-out of discoverability features like the profile directory. This flag may also be used as an indicator of the user's preferences toward being included in external discovery services, such as search engines or other indexing tools. If you are implementing such a tool, it is recommended that you respect this property if it is present. This is implemented using an extra property `discoverable` on objects.
```

Comment:
the description here is unchanged since 2019, i just moved it up from lower in the page (red line 521 in this changeset). 

aside from that, it is worth noting that the current definition also says it is applied on "objects" without specifying that mastodon only applies it on actors. the wording is indeed vague, but i believe the original mention of "search engines and indexing tools" was just a generalization of the profile directory (which is an "internal" discovery service). you could envision a "search for profiles" service that would let you find discoverable accounts in a way similar to the profile directory, but with more details like including their bio and featured hashtags. but it doesn't really make sense to define this property only for actors even if it only gets used on actors.

the real contention to me is in whether this account-level preference carries over to posts. i think it generally does not, and therefore i avoided saying this (although i can see how it is still vague). relatedly, this is why i think `discoverable` should be attached to posts as well: to remove that kind of ambiguity. if we attach `discoverable` at the post level, you no longer need to check the account property. regardless, we might still (may or may not) consider a policy of deferring to the account property if the post doesn't declare whether it is `discoverable`. this basically creates three or four states:

- if the post is `discoverable: true`, it SHOULD be shown in discovery features
- if the post is `discoverable: false`, it MUST NOT be shown in discovery features
- if the post is `discoverable: null` or the property is missing:
  - if the account is `discoverable: true`, the post MAY be shown in discovery features
  - if the account is `discoverable: false`, the post SHOULD NOT be shown in discovery features
  - if the account is `discoverable: null` or the property is missing, the post MAY be shown in discovery features if some other mechanism establishes consent, but SHOULD NOT be shown in lieu of any other mechanism